src/app/components/core/developer/additional-services/additional-service-modal/additional-service-modal.component.ts

```diff
@@ -369,16 +369,18 @@ export class AdditionalServiceModalComponent {
             throw response;
           }
         }),
-        retryWhen(result =>
-          result.pipe(
+        retryWhen(result =>     
+          result.pipe(           
             // only if a server returned an error, stop trying and pass the error down
             tap(serviceStatus => {
               if (serviceStatus.error) {
                 this.ngProgress.done();
                 this.toggleModal(false);
                 this.reenableButton.emit(false);
                 this.additionalServicesUtils.navToAdditionalServicePage(this.fromListPage, this.serviceInfo.process);
-                throw serviceStatus.error;
+                if ((status !== 'addService') || (status === 'addService' && i > 3)) {
```

Comment:
why `i>3`?  what is the purpose of this check?

Reply:
If service is being added, we don't get the added service in a very first/second API call and it throws error that 'service is not found'. So I added a check so that it will through error in the last attempt.